event-details/EventDetailsLoading.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  DefinedQueryObserverResult,
+  UseQueryResult,
+  useQuery
+} from "@tanstack/react-query"
+import { BlockedEvent, CurrentUserEvent } from "./Event"
+
+/**
+ * A result from loading a single event for the details screen.
+ */
+export type EventDetailsLoadingResult =
+  | { status: "not-found" | "deleted" }
+  | { status: "blocked"; event: BlockedEvent }
+  | { status: "success"; event: CurrentUserEvent }
+
+export type UseLoadEventDetailsResult =
+  | { status: "loading" | "not-found" | "deleted" }
```

Comment:
May want to use this type for other queries, but we can wait until more integration

Reply:
For sure